### ISSUE TITLE
Namespaces.ReferenceUsedNamesOnly & Throwable

### ISSUE THREAD

**Issue by fprochazka** [slevomat/coding-standard#41]
Hi, can you tell me please what is the best way to solve the following problem? We want to check if `Throwable` is used absolutely in catch statements, same as other exceptions.

I know I can solve it simply using the `specialExceptionNames` property like this

```xml
<rule ref="SlevomatCodingStandard.Namespaces.ReferenceUsedNamesOnly">
	<properties>
		<property name="specialExceptionNames" type="array" value="
			Throwable
		"/>
	</properties>
</rule>
```

But wouldn't it be cleaner to have a condition for it in the sniff? Since `Throwable` is not really a "special case" it should imho have same behaviour as other exceptions and so should [the other new exceptions](https://secure.php.net/manual/en/reserved.exceptions.php) in PHP7.

What do you think? How should this be solved? I'd like to discuss this before I send a PR and spend time writing tests.

**Comment by ondrejmirtes** [comment]
I think it's not enough to change just ReferenceUsedNamesOnly, but also FullyQualifiedExceptionsSniff.

There is a clear omission in these sniffs that didn't occur to me once we started using PHP 7 - because no one ever put Throwable in a use statement here 😄 

Adding mention about `Throwable` to ReferenceUsedNamesOnly will help when `allowFullyQualifiedExceptions` is set to false.

Adding `Throwable` to FullyQualifiedExceptionsSniff will mark `catch (Throwable $e)` as an error which is your desired behaviour.

You have to do both to have consistent behaviour.

Feel free to open a PR (with tests), otherwise I will do it in a few days and merge it into master (no reason to wait until 2.0 with full PHP 7 support, because this is backwards-compatible).

**Comment by fprochazka** [comment]
Thanks for the hint! I have something like this now.

```xml
	<rule ref="SlevomatCodingStandard.Namespaces.FullyQualifiedExceptions">
		<properties>
			<property name="specialExceptionNames" type="array" value="
				Throwable
			"/>
		</properties>
	</rule>
	<rule ref="SlevomatCodingStandard.Namespaces.ReferenceUsedNamesOnly">
		<properties>
			<property name="allowFullyQualifiedExceptions" value="true"/>
			<property name="specialExceptionNames" type="array" value="
				Throwable
			"/>
		</properties>
	</rule>
```

I wanted to send the PR myself, but got confused by your tests. If you'd have time to solve this, that would be great :)

**Comment by ondrejmirtes** [comment]
Can you verify that PR #42 fixes your issue without the need for special configuration mentioning `Throwable`? Thanks.

**Comment by ondrejmirtes** [comment]
It just occured to me that I should implement checking for types ending with `Error` too.

**Comment by ondrejmirtes** [comment]
Alright, updated.

**Comment by ondrejmirtes** [comment]
I merged the PR #42: https://github.com/slevomat/coding-standard/commit/d2989ad1184e750570010bbfa89a2b54da7208f8

**Comment by VasekPurchart** [comment]
Thanks! :)
